lib/smb2-cmd-set-info.c

```diff
@@ -92,6 +93,35 @@ smb2_encode_set_info_request(struct smb2_context *smb2,
 
                         eofi = req->input_data;
                         smb2_set_uint64(iov, 0, eofi->end_of_file);
+                        break;
+                case SMB2_FILE_RENAME_INFORMATION:
+                        rni = req->input_data;
+
+                        struct ucs2 *name = utf8_to_ucs2(rni->file_name);
+                        if (name == NULL) {
+                                smb2_set_error(smb2, "Could not convert name into UCS2");
+                                return -1;
+                        }
+
+                        len = 20 + name->len * 2;
+                        smb2_set_uint32(iov, 4, len); /* buffer length */
+
+                        buf = malloc(len);
+                        if (buf == NULL) {
+                                smb2_set_error(smb2, "Failed to allocate set "
+                                               "info data buffer");
+                                return -1;
```

Comment:
free(name)) here ?

Reply:
my fault. will fix

Reply:
I have merged it. I will fix it in a small followup patch